skstore/ts/src/skstore_api.ts

```diff
@@ -0,0 +1,588 @@
+   */
+  multimapReduce<
+    K1 extends TJSON,
+    V1 extends TJSON,
+    K2 extends TJSON,
+    V2 extends TJSON,
+    V3 extends TJSON,
+  >(
+    mappings: Mapping<K1, V1, K2, V2>[],
+    accumulator: Accumulator<V2, V3>,
+  ): EHandle<K2, V3>;
+
+  /**
+   * Creates a lazy reactive map attched to a asynch call
+   * @param get - the function the gather the values from others reactive maps
+   * @param call - the async function to call with gathered values
+   * @returns {LHandle} The the resulting lazy reactive map handle
+   */
+  asyncLazy<K extends TJSON, V extends TJSON, P extends TJSON, M extends TJSON>(
+    get: (key: K) => P,
+    call: (key: K, params: P) => Promise<AValue<V, M>>,
+  ): LHandle<K, Loadable<V, M>>;
+
+  log(object: TJSON): void;
+}
+
+export interface Handles {
+  register(v: any): int;
+  get(id: int): any;
+  apply<T>(id: int, parameters: T[]): T;
+  delete(id: int): any;
+  name(metadata: Metadata): string;
+}
+
+export interface FromWasm {
+  // Handle
+  SKIP_SKStore_map(ctx: ptr, eagerHdl: ptr, name: ptr, fnPtr: int): ptr;
+
+  SKIP_SKStore_mapReduce(
+    ctx: ptr,
+    eagerHdl: ptr,
+    name: ptr,
+    fnPtr: int,
+    accumulator: int,
+    accInit: ptr,
+  ): ptr;
+
+  SKIP_SKStore_get(ctx: ptr, getterHdl: ptr, key: ptr): ptr;
+  SKIP_SKStore_getFromTable(ctx: ptr, table: ptr, key: ptr, index: ptr): ptr;
+
+  SKIP_SKStore_maybeGet(ctx: ptr, getterHdl: ptr, key: ptr): ptr;
+
+  SKIP_SKStore_getLazy(ctx: ptr, lazyId: ptr, key: ptr): ptr;
+
+  SKIP_SKStore_getSelf(ctx: ptr, selfHdl: ptr, key: ptr): ptr;
+
+  SKIP_SKStore_set(ctx: ptr, inputHdl: ptr, key: ptr, value: ptr): void;
+  SKIP_SKStore_remove(ctx: ptr, inputHdl: ptr, key: ptr): void;
+  SKIP_SKStore_size(ctx: ptr, eagerHdl: ptr): number;
+
+  SKIP_SKStore_toSkdb(ctx: ptr, eagerHdl: ptr, table: ptr, fnPtr: int): void;
+
+  // NonEmptyIterator
+  SKIP_SKStore_iteratorNext(it: ptr): Opt<ptr>;
+  SKIP_SKStore_iteratorUniqueValue(it: ptr): Opt<ptr>;
+  SKIP_SKStore_iteratorFirst(it: ptr): ptr;
+  // Writer
+  SKIP_SKStore_writerSetValues(writer: ptr, key: ptr, size: int): void;
+  SKIP_SKStore_writerSet(writer: ptr, key: ptr, value: ptr): void;
+  SKIP_SKStore_writerRemove(writer: ptr, key: ptr): void;
+
+  // Store
+  SKIP_SKStore_createFor(session: ptr): float;
+
+  // SKStore
+  SKIP_SKStore_input(ctx: ptr, name: ptr, values: ptr): ptr;
+  SKIP_SKStore_asyncLazy(ctx: ptr, name: ptr, paramsFn: int, lazyFn: int): ptr;
+  SKIP_SKStore_lazy(ctx: ptr, name: ptr, lazyFn: int): ptr;
+  SKIP_SKStore_fromSkdb(ctx: ptr, table: ptr, name: ptr, fnPtr: int): ptr;
+  SKIP_SKStore_multimap(ctx: ptr, name: ptr, mappings: ptr): ptr;
+  SKIP_SKStore_asyncResult(
+    callId: ptr,
+    name: ptr,
+    key: ptr,
+    param: ptr,
+    value: ptr,
+  ): number;
+
+  SKIP_SKStore_multimapReduce(
+    ctx: ptr,
+    name: ptr,
+    mappings: ptr,
+    accumulator: int,
+    accInit: ptr,
+  ): ptr;
+}
+
+export interface SKJSON extends Shared {
```

Comment:
What's going on with this interface / how does it relate to the code in `skstore_skjson.ts`?  As far as I can see this is defined but never actually implemented.